### PR TITLE
dev(hansbug): add CI support && fix font missing problem && add font installation guide in README.md

### PR DESCRIPTION
**Pull request by HansBug**
* [x] 添加CI支持，现已可以直接编译并提供下载
* [x] 修复Linux环境自动编译情况下的字体缺失问题
* [x] 在`README.md`中补充Linux环境下字体安装方法

### CODE REVIEW

**Comment by HansBug** [comment]
#294

**Comment by HansBug** [comment]
在fork仓库这边，CI已经通过，速度完全可接受，并可以在artifacts处下载：https://github.com/HansBug/BUAAthesis/actions/runs/1539116095

**Review by sighingnow — APPROVED** [review]

**Comment by sighingnow** [comment]
LGTM. Besides the custom font part.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          fetch-depth: 10
+      - name: Set up system dependences on linux
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y tree cloc make
+      - name: Load the font requirements
+        run: |
+          git clone "https://github.com/HansBug/BUAAthesis_fonts" -b master local_fonts
```
Comment (sighingnow) [inline .github/workflows/build.yml:34]: How about download the fonts from

```
wget https://dl.freefontsfamily.com/download/Times-New-Roman-Font/ -o times-new-roman.zip
wget https://www.wfonts.com/download/data/2014/06/01/stxingkai/stxingkai.zip
```

Comment (sighingnow) [reply .github/workflows/build.yml:34]: We could add a link to the readme to tell users that they can download those two fonts from above links.

File: README.md [hunk 2]
```diff
@@ -39,6 +39,23 @@
 
 对于老版本的 LaTeX 发行版，请通过包管理器升级 ctex 的版本。
 
+
+
+此外，对于非Windows环境，会需要额外安装两个字体：
+
+* `Times New Roman`，该字体可以在Windows系统字体库中获取，也可以在Ubuntu/Debian下快速安装
```
Comment (sighingnow) [inline README.md:46]: Then we prefer to install times new roman using apt-get, rather than download it.

Comment (sighingnow) [reply README.md:46]: > 

In CI actions.

File: .github/workflows/build.yml (hunk 1 above)
Comment (HansBug) [reply .github/workflows/build.yml]: Completed.

File: README.md (hunk 2 above)
Comment (HansBug) [reply README.md]: Use the link to download temporarily, and then change to apt-get if you really need it.

**Review by sighingnow — APPROVED** [review]
I'm going to merge this one as no opposition

**Comment by sighingnow** [comment]
Thank you @HansBug ❤️

**Pull request merged**